### PR TITLE
Fixing some crashing problems

### PR DESCRIPTION
**Pull request by dmchong**
added some error checking and used autopep8 aggressive... 

autopep8 --in-place --aggressive --aggressive <filename>

### CODE REVIEW

File: lunchbot.py [hunk 1]
```diff
@@ -85,29 +93,26 @@ def weighted_choice(self, choices):
...
+
+            # error checking.
+            if restaurant is None:
+                return "Please specify a restaurant"
+            if weight is None:
+                return "Please specify a weight"
+            else:
+                try:
+                    weight = int(weight)
+                except:
```
Comment (n-martin) [inline lunchbot.py]: We should be more specific with regards to the exceptions we are trying to catch

File: lunchbot.py [hunk 2]
```diff
@@ -133,7 +139,13 @@ def handle_lunchbot_command(self, command, user):
             # Update data.json file and reload global variables
             self.update_data()
             self.read_data()
-        elif 'remove restaurant' in command.lower():
+        except:
+            msg = 'please try again'
```
Comment (n-martin) [inline lunchbot.py]: Messages should start capitalized :)

File: lunchbot.py [hunk 3]
```diff
@@ -154,13 +166,44 @@ def handle_lunchbot_command(self, command, user):
...
+            msg = self.HELPTEXT
+        elif 'list restaurants' in command.lower():
+            self.post_ephemeral = True
+            msg = self.BLANK
+            for restaurant in self.RESTAURANTS:
+                msg = '\n'.join([msg, '%s has weight %s' % (restaurant['name'], restaurant['weight'])])
+
+        elif 'add restaurant' in command.lower():
+            msg = self.add_restaurant_command(command)
+
```
Comment (n-martin) [inline lunchbot.py]: Extra lines here between elifs

File: lunchbot.py [hunk 4]
```diff
@@ -26,6 +26,14 @@ class Lunchbot(object):
 
     def __init__(self):
         self.read_data()
+        self.post_ephemeral = False
+
+        # String Constants
+        self.HELPTEXT = ('Here are the actions that Lunchbot can perform:\n'
```
Comment (n-martin) [inline lunchbot.py]: These constants could be moved out with the others.

File: lunchbot.py [hunk 5]
```diff
@@ -85,29 +93,26 @@ def weighted_choice(self, choices):
...
             # Split command but keep quoted substrings as a single unit
             pieces = [p for p in re.split("( |\\\".*?\\\"|'.*?')", command) if p.strip()]
+
             # Get restaurant and weight to add from command
             restaurant = pieces[-2]
             weight = int(pieces[-1])
+
+            # error checking.
+            if restaurant is None:
+                return "Please specify a restaurant"
```
Comment (n-martin) [inline lunchbot.py]: I think these error messages should be posted ephemerally

Comment (dmchong) [reply lunchbot.py]: I was just thinking of that right after i requested the pull lol....

**Review by dmchong — COMMENTED** [review]
fixing suggested changes.

File: lunchbot.py [hunk 6]
```diff
@@ -20,12 +20,22 @@ class Lunchbot(object):
...
+                 'add restaurant <name> <weight>\n'
+                 'remove restaurant <name>\n')
 
     def __init__(self):
         self.read_data()
+        self.post_ephemeral = False
+
+        # String Constants
+
+        self.BLANK = ""
```
Comment (n-martin) [inline lunchbot.py]: This could be moved out as well! The comment could be removed as well

File: lunchbot.py [hunk 7]
```diff
@@ -133,7 +144,14 @@ def handle_lunchbot_command(self, command, user):
...
             self.update_data()
             self.read_data()
-        elif 'remove restaurant' in command.lower():
+        except Exception as e:
+            self.post_ephemeral = True
+            msg = 'Please try again.\n%s' % e
+        return msg
+
+    def remove_restaurant_command(self, command):
+
```
Comment (n-martin) [inline lunchbot.py]: Extra line here

File: lunchbot.py [hunk 8]
```diff
@@ -154,13 +172,41 @@ def handle_lunchbot_command(self, command, user):
...
+        elif 'list restaurants' in command.lower():
+            self.post_ephemeral = True
+            msg = self.BLANK
+            for restaurant in self.RESTAURANTS:
+                msg = '\n'.join([msg, '%s has weight %s' % (restaurant['name'], restaurant['weight'])])
+        elif 'add restaurant' in command.lower():
+            msg = self.add_restaurant_command(command)
+        elif 'remove restaurant' in command.lower():
+            msg = self.remove_restaurant_command(command)
+        elif '' in command.lower():
```
Comment (n-martin) [inline lunchbot.py:196]: What the difference between this case and the else case?

File: lunchbot.py [hunk 9]
```diff
@@ -154,13 +172,41 @@ def handle_lunchbot_command(self, command, user):
...
+            self.post_ephemeral = True
+            msg = 'What can i do for you? Please type "Lunchbot help" to see what I can do :)'
         else:
-            post_ephemeral = True
+            self.post_ephemeral = True
             msg = 'I\'m not smart enough to understand what you mean! Please type "Lunchbot help" to see what I can do :)'
 
-        if post_ephemeral:
+        # posting messages
+        # todo DMing someone instead of lunch chat.
```
Comment (n-martin) [inline lunchbot.py]: We can remove this todo, there is already an issue raised for it!

**Review by n-martin — CHANGES_REQUESTED** [review]
Looking good, thanks for the fixes Douglas. Just a few minor things to fix and then it should be good to merge!

File: lunchbot.py (hunk 8 above)
Comment (dmchong) [reply lunchbot.py:196]: this is if its a blank...

like lunchbot <blank>
else if its anything other than blank

Comment (dmchong) [reply lunchbot.py:196]: lucnhbot sushi

**Review by dmchong — COMMENTED** [review]
removed the recommended stuff to remove

File: lunchbot.py [hunk 10]
```diff
@@ -154,13 +167,40 @@ def handle_lunchbot_command(self, command, user):
...
+        Takes a command from Slack chat directed at Lunchbot and performs the action if it is supported
+        """
+        self.post_ephemeral = False
+
+        if 'help' in command.lower():
+            self.post_ephemeral = True
+            msg = self.HELP_TEXT
+        elif 'list restaurants' in command.lower():
+            self.post_ephemeral = True
+            msg = self.BLANK
```
Comment (n-martin) [inline lunchbot.py:189]: I can't find self.BLANK anywhere in this file anymore, did you remove it completely?

**Review by n-martin — COMMENTED** [review]
I don't think we need to include vcs.xml in .idea. Is that a config file for your IDE?

**Review by dmchong — COMMENTED** [review]
that blank >_<

File: lunchbot.py (hunk 10 above)
Comment (dmchong) [reply lunchbot.py:189]: >_< added it back in

**Review by dmchong — COMMENTED** [review]
removed that xml

**Review by dmchong — COMMENTED** [review]
removed slackbot.py

**Review by n-martin — APPROVED** [review]
Nice, looks good to me!

File: lunchbot.py [hunk 11]
```diff
@@ -85,29 +92,29 @@ def weighted_choice(self, choices):
...
+            else:
+                try:
+                    weight = int(weight)
+                except ValueError:
+                    self.post_ephemeral = True
+                    return "Enter a valid weight"
+
             # Remove quotations from restaurant string if there are any
             if restaurant[0] == '"':
                 quotation = '"'
```
Comment (ChristopherzwTan) [inline lunchbot.py:120]: Quote checking even, very nice... @

**Review by ChristopherzwTan — APPROVED** [review]
Ok, looks pretty good. We;'re doing it live!

File: lunchbot.py (hunk 11 above)
Comment (dmchong) [reply lunchbot.py:120]: wasn't me.... i just left it in there... lol

**Comment by n-martin** [comment]
Here we go! Best of luck Lunchbot.

**Pull request merged**